edifact/models/tests/test_edifact.py

```diff
@@ -0,0 +1,61 @@
+import unittest
+from datetime import datetime
+
+from edifact.models.edifact import Edifact
+from edifact.models.interchange import InterchangeHeader, InterchangeTrailer
+from edifact.models.message import MessageHeader, MessageTrailer, BeginningOfMessage, NameAndAddress, DateTimePeriod, \
+    ReferenceTransactionType, ReferenceTransactionNumber, SegmentGroup
+from edifact.models.tests.segments_comparison_util import SegmentComparisonTest
+
+SEGMENT_COMPARISON = SegmentComparisonTest()
+
+EDI_FILE = """UNB+UNOA:2+GP123+HA456+200427:1737+00000045'
+UNH+00000056+FHSREG:0:1:FH:FHS001'
+BGM+++507'
+NAD+FHS+HA456:954'
+DTM+137:202004271737:203'
+RFF+950:G1'
+S01+1'
+RFF+TN:5174'
+UNT+8+00000056'
+UNZ+1+00000045'"""
+
+interchange_header = InterchangeHeader('GP123', 'HA456', datetime(2020, 4, 27, 17, 37), 45)
+interchange_trailer = InterchangeTrailer(1, 45)
+message_header = MessageHeader(56)
+message_trailer = MessageTrailer(8, 56)
+beginning_of_message = BeginningOfMessage()
+name_and_address = NameAndAddress(NameAndAddress.QualifierAndCode.FHS, 'HA456')
+date_time_period = DateTimePeriod(DateTimePeriod.TypeAndFormat.TRANSLATION_TIMESTAMP, datetime(2020, 4, 27, 17, 37))
+reference_transaction_type = ReferenceTransactionType(ReferenceTransactionType.TransactionType.ACCEPTANCE)
+reference_transaction_number = ReferenceTransactionNumber(5174)
+segment_group = SegmentGroup(1)
+
+
+class TestEdifact(unittest.TestCase):
+
+    def test_create_from_message(self):
+        edifact = Edifact.create_edifact_from_message(EDI_FILE)
+        # TODO: validate parsed segments
```

Comment:
how do we know this code works? does the validation of the parsed segments need to be here to verify this?

Reply:
this TODO is to be removed. You have segments validation. below

Reply:
removed this todo, the validation is below